browser/ui/tabs/split_view_browser_data_observer.h

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_TABS_SPLIT_VIEW_BROWSER_DATA_OBSERVER_H_
+#define BRAVE_BROWSER_UI_TABS_SPLIT_VIEW_BROWSER_DATA_OBSERVER_H_
+
+#include "base/observer_list_types.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+
+class SplitViewBrowserDataObserver : public base::CheckedObserver {
+ public:
+  virtual void OnTileTabs(const SplitViewBrowserData::Tile& tile) = 0;
+  virtual void OnWillBreakTile(const SplitViewBrowserData::Tile& tile) = 0;
+
+ protected:
+  SplitViewBrowserDataObserver() = default;
```

Comment:
Just curious - is there a reason we add a constructor? I think in other places I've seen `CheckedObserver` being implemented they  just provide a destructor. In fact, I see a good chunk don't even implement an override for the destructor.

Reply:
Yeah, it's intended to control the access to `protected`. That's because "Observer" shouldn't be created without the concrete class, but in case all methods are the `pure virtual`, then the `protected contsturctor` doesn't have point.

Regarding the destructor, IMO observer interface has the protected desturctor as the object references the observer must not delete it. This makes the interface verbose, but as ownership and lifecycle is important in C++ and there's nothing like `interface` or `abstract` keyword in Java, I believe this was recommended back then.

Reply:
Cool thanks for the explanation! I'll make sure I always add the destructor for my observer classes!

Reply:
Sorry, I should admit that my old habbits are not necessary anymore as the CheckedObserver declared its destructor `public` already. Will remove ctor and detor